malinka.el

```diff
@@ -688,20 +710,41 @@ Compatible means that it's of a big enough version in order to be able to genera
 		 (:else ;; major version being 1 means not supported
 		  nil))))))
 
-(defun malinka--cmake-create-compiledb (project-map)
-  "Create a compilation database for a PROJECT-MAP using CMAKE."
-  (let* ((configure-cmd (malinka--project-configure-cmd project-map))
-         (build-dir (malinka--project-build-directory project-map))
-         (nbuild-cmd (format "cd %s && %s -DCMAKE_EXPORT_COMPILE_COMMANDS=ON" build-dir configure-cmd))
+(defun malinka--project-contains-compile-db-cmd? (project-map)
+  "Detect if the malinka PROJECT-MAP contains none empty compile-db-cmd."
+  (not (s-blank? (malinka--project-compile-db-cmd project-map))))
+
+(defun malinka--build-cmd-for-compiledb (project-map project-type)
+  "Generate correct compile-db build cmd for different project type.
+
+Current support PROJECT-TYPE are: compile-db-cmd, cmake, bear."
+  (let* ((build-dir (malinka--project-build-directory project-map))
+         (compile-db-cmd (malinka--project-compile-db-cmd project-map))
+         (configure-cmd (malinka--project-configure-cmd project-map)))
+    (cond ((s-equals? project-type "compile-db-cmd")
+           (format "cd %s && %s" build-dir compile-db-cmd))
+          ((s-equals? project-type "cmake")
+           (format "cd %s && %s -DCMAKE_EXPORT_COMPILE_COMMANDS=ON" build-dir configure-cmd))
+          ((s-equals? project-type "bear")
+           (format "cd %s && bear -a %s" build-dir configure-cmd))
```

Comment:
Oh, sorry, I missed this one.

It still would be an issue if user only provides `configure-cmd: make` with "bear" installed. 

- I think empty compile database checking in `malinka--handle-bear-finish` function would relief the situation. 

- Another way is removing `bear` type project support and let user decides how to generated compilation database (by providing `compile-db-cmd`).

What do you think?